### PR TITLE
JI:773008 P2022-2092 (language selector): fix language detection bug

### PR DESCRIPTION
**Pull request by piotrrawski**
resolve problem with detection and translation of regional languages

### CODE REVIEW

**Comment by Reden4005** [comment]
Wyświetla się nieprawidłowy querystring.
Myślę, że to domknie sprawę:

`export const isOtherLanguage = (lang: string) => {
  return lang.slice(0, 2) !== SupportedLanguage.En;
};`

File: libs/shared/src/utils/load-i18n.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ export const loadI18n = ({ env, ns }: I18nOptions) => {
     .init({
       fallbackLng: "en",
       supportedLngs: [SupportedLanguage.En, SupportedLanguage.Pl],
-      load: "currentOnly",
+      load: "languageOnly",
       debug: env === "development",
 
       resources: translationResources,
```
Comment (Reden4005) [inline libs/shared/src/utils/load-i18n.ts:37]: Wyświetla się nieprawidłowy querystring.
Myślę, że to domknie sprawę:

export const isOtherLanguage = (lang: string) => { return lang.slice(0, 2) !== SupportedLanguage.En; };

Comment (kacperChlon) [reply libs/shared/src/utils/load-i18n.ts:37]: and avoid magic numbers <-readme.md

File: libs/shared/src/utils/lang-fns.tsx [hunk 2]
```diff
@@ -16,5 +16,5 @@ export const isDefaultLanguage = (lang: string) => {
 };
 
 export const isOtherLanguage = (lang: string) => {
-  return lang !== SupportedLanguage.En;
+  return lang.slice(0, 2) !== SupportedLanguage.En;
```
Comment (kacperChlon) [inline libs/shared/src/utils/lang-fns.tsx:19]: and avoid magic numbers <-readme.md

Comment (piotrrawski) [reply libs/shared/src/utils/lang-fns.tsx]: done

File: libs/shared/src/utils/load-i18n.ts (hunk 1 above)
Comment (piotrrawski) [reply libs/shared/src/utils/load-i18n.ts:37]: done

File: libs/shared/src/types/translations.ts [hunk 3]
```diff
@@ -8,3 +8,8 @@ export enum TranslationNamespace {
   Common = "common",
   Ui = "ui"
 }
+
+export enum SliceRange {
```
Comment (kacperChlon) [inline libs/shared/src/types/translations.ts:12]: what slice? ;) I could use this name in 1000 files without knowing what is this slice. 

please delete it and add this:

File: libs/shared/src/utils/lang-fns.tsx [hunk 4]
```diff
@@ -16,5 +16,5 @@ export const isDefaultLanguage = (lang: string) => {
 };
```
Comment (kacperChlon) [inline libs/shared/src/utils/lang-fns.tsx:17]: const LANG_FIRST_LETTER_POSITION = 0
const LANG_LAST_LETTER_POSITION = 2

Comment (piotrrawski) [reply libs/shared/src/utils/lang-fns.tsx:18]: done

**Review by Reden4005 — COMMENTED** [review]
LGTM

File: libs/shared/src/utils/lang-fns.tsx (hunk 4 above)
Comment (kacperChlon) [reply libs/shared/src/utils/lang-fns.tsx:18]: We are using them inside components not in shared directory, there is no need to put those constants into the shared common constants, we won't use them in any other place than lang-fns. Please put those constants within lang-fns

Comment (piotrrawski) [reply libs/shared/src/utils/lang-fns.tsx:19]: make sense, done

**Review by aniasgit — COMMENTED** [review]
LGTM

File: libs/shared/src/utils/lang-fns.tsx [hunk 5]
```diff
@@ -2,6 +2,8 @@ import { i18n } from "i18next";
 
 import { SupportedLanguage } from "../types";
 
+export const LANG_FIRST_LETTER_POSITION = 0;
+export const LANG_LAST_LETTER_POSITION = 2;
 export interface ChangeLanguage {
```
[the remark this answers is not in the document]
Comment (piotrrawski) [reply libs/shared/src/utils/lang-fns.tsx:9]: done

**Comment by krysia1** [comment]
LGTM 💃

**Review by pawelwojdas — COMMENTED** [review]
LGTM

**Review by kacperChlon — APPROVED** [review]

**Review by aniasgit — APPROVED** [review]

**Pull request merged**